src/enclave/curl.h

```diff
@@ -89,25 +91,23 @@ namespace afetch {
       curl_easy_setopt(curl, CURLOPT_CAPATH, NULL);
 
       auto res = curl_easy_perform(curl);
-
       if (res != CURLE_OK) {
-        const char* err_s = curl_easy_strerror(res);
-        throw CurlOtherError(url, err_s, res);
+        response.error_message = curl_easy_strerror(res);
+        if (verbose) {
+          std::cerr << "Fetch failed: " << response.error_message << std::endl;
+        }
+        response.status = NULL;
```

Comment:
`status` is not a pointer, so you shouldn't be assigning `NULL` to it (I'm kind of amazed this even compiles). This is just equivalent to assigning 0 but more confusing.

Either use 0, or make it a `std::optional<int64_t>`

Reply:
Alternatively, don't return a `Response` when an error happens: keep throwing a `CurlError` (might need to change its definition slightly), and add a try-catch around `auto response = curl.fetch(url);` in `enclave.cpp`

Reply:
Thanks I wanted to ask about using NULL there, I've gone with the try-catch